src/project/ProjectModel.js

```diff
@@ -1240,6 +1254,50 @@ define(function (require, exports, module) {
     };
 
     /**
+     * Moves the item in oldPath to the newDirectory directory
+     * @param {string} oldPath  old path of the item
+     * @param {string} newDirectory path of the directory to move the item
+     * @return {$.Promise} promise resolved when the item is moved
+     */
+    ProjectModel.prototype.moveItem = function(oldPath, newDirectory) {
+        var self = this,
+            d = new $.Deferred(),
+            fileName = FileUtils.getBaseName(oldPath),
+            fullNewPath = newDirectory + fileName;
+
+        // Add trailing slash if directory is moved
+        if (!_pathIsFile(oldPath)) {
+            fullNewPath = _ensureTrailingSlash(fullNewPath);
+        }
+
+        // Reusing the _renameItem for moving item
+        this._renameItem(oldPath, fullNewPath, fileName, !_pathIsFile(fullNewPath)).then(function () {
+            var newDirectoryRelative = self.makeProjectRelativeIfPossible(newDirectory),
+                needsLoading    = !self._viewModel.isPathLoaded(newDirectoryRelative);
```

Comment:
Maybe save the reference to `var viewModel = self._viewModel` to make the calls a bit less terse.